CMakeLists.txt

```diff
@@ -552,6 +553,16 @@ if(WITH_SYSTEM_KLT)
     find_package(KLT REQUIRED)
 endif()
 
+set(JXL_LIBS "libjxl libjxl_threads")
+if(WITH_JXL)
+    pkg_check_modules(JXL REQUIRED IMPORTED_TARGET ${JXL_LIBS})
+else()
+    pkg_check_modules(JXL IMPORTED_TARGET ${JXL_LIBS})
+endif()
+if(JXL_FOUND)
+    add_definitions(-DLIBJXL)    
+endif()
```

Comment:
Not a fan of this logic... If the user is explicitly expected to set WITH_JXL to ON, shouldn't silently do stuff behind their back when they no longer don't - they'll still get it built when they change it back to OFF just because the library is present on their system.

In fact, it might be better to default this to ON, and enable the user to explicitly turn it OFF completely then.

Reply:
Auto detect is needed for compatibility with workflows that may not have access to libjxl.

Forced on prevents people from wasting their time building without libjxl.

Forced off is left out until I figure out how to implement a tristate because no reasonable person would bother downloading a PR with the intention of disabling it during build.